BeeminderWidget/BeeminderWidgetConfigurationIntents.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  BeeminderWidgetConfigurationIntents.swift
+//  BeeminderWidget
+//
+//  Created by krugerk on 2024-11-13.
+//
+
+import AppIntents
+import BeeKit
+import WidgetKit
+
+struct GoalBasicsConfigurationAppIntent: WidgetConfigurationIntent {
+    static let title: LocalizedStringResource = "Goal Basics"
+    static let description = IntentDescription("Shows basic data of a goal!")
+
+    @Parameter(title: "Goal Name (aka slug)",
+               default: "steps",
+               inputOptions: String.IntentInputOptions(keyboardType: .default, capitalizationType: .none, autocorrect: false))
+    var goalName: String
+}
+
+struct PledgedConfigurationAppIntent: WidgetConfigurationIntent {
+    static let title: LocalizedStringResource = "Amount Pledged"
+    static let description = IntentDescription("Shows the sum you currently have pledged!")
+
+    enum BeeminderPledgeDenomination: String, AppEnum {
+        case honeyMoney, usDollar
+
+        static let typeDisplayRepresentation: TypeDisplayRepresentation = "Pledge Denomination"
+        static let caseDisplayRepresentations: [BeeminderPledgeDenomination: DisplayRepresentation] = [
+            .honeyMoney: "Honey Money",
+            .usDollar: "US Dollar",
+        ]
+    }
+
+    @Parameter(title: "Denomination", default: .honeyMoney)
+    var denomination: BeeminderPledgeDenomination
+}
+
+struct GoalCountdownConfigurationAppIntent: WidgetConfigurationIntent {
+    static let title: LocalizedStringResource = "Colored Goal"
+    static let description = IntentDescription("Shows a goal in its countdown color!")
+
+    @Parameter(title: "Goal Name (aka slug)",
+               default: "dial",
+               inputOptions: String.IntentInputOptions(keyboardType: .default, capitalizationType: .none, autocorrect: false))
+    var goalName: String
```

Comment:
currently requires user to manually type in the goal name. 

- [ ] provide a dropdown list of the goals

Reply:
initial implementation in https://github.com/beeminder/BeeSwift/pull/515/commits/94262a34c28d197d483ad4b4dcf30b49e495be2e